dvc/external_repo.py

```diff
@@ -61,68 +69,186 @@ def clean_repos():
         _remove(path)
 
 
-class ExternalRepo(Repo):
-    def __init__(self, root_dir, url):
-        super().__init__(root_dir)
-        self.url = url
-        self._set_cache_dir()
-        self._fix_upstream()
+class BaseExternalRepo:
+    @cached_property
+    def repo_tree(self):
+        return RepoTree(self)
 
-    def pull_to(self, path, to_info):
+    # temporary hack to make cache use WorkingTree and not GitTree, because
+    # cache dir doesn't exist in the latter.
+    @contextmanager
+    def fix_cache_tree(self):
+        saved_tree = self.tree
+        self.tree = WorkingTree(self.root_dir)
+        yield
+        self.tree = saved_tree
+
+    def get_external(self, path, to_info, jobs=None):
```

Comment:
this function is still mostly just old `pull_to` but what it should eventually do is more like

1. fetch needed outs from `path` into cache (`self.fetch_external([path])`)
2. `self.repo_tree.walk(path)`
3. use `repo_tree.open()` to copy (from fobj's) files from either git index or cache to `to_info`